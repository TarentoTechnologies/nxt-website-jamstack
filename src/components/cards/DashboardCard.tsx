import { Link } from "gatsby";
import { GatsbyImage, getImage } from "gatsby-plugin-image";
import React from "react";

import forwardArrow from "../../../static/icons/arrow_forward_black.svg";
import {
  cta1,
  planHeading,
  subText2,
} from "../../styles/style-guide/Typography.module.css";
import { dashboardCard } from "./DashboardCard.module.css";
import { ctaIcon } from "./PortfolioCard.module.css";
import { ctaIconPos } from "./ShowcaseCard.module.css";

interface DashboardCardProps {
  title: string;
  description: string;
  imgSrc: any;
  imgAltText?: string;
  ctaLabel: string;
  ctaLink: string;
  isCTAExternal: boolean;
}

export const DashboardCard = ({
  title,
  description,
  imgSrc,
  imgAltText,
  ctaLabel,
  ctaLink,
  isCTAExternal,
}: DashboardCardProps) => {
  return (
    <div className={`${dashboardCard} whiteBg position-relative`}>
      <div data-aos="fade-up">
        <GatsbyImage
          image={
            getImage(imgSrc?.localFile)
              ? getImage(imgSrc?.localFile)
              : imgSrc?.localFile
          }
          alt={imgAltText ? imgAltText : ""}
        />
      </div>
      <div className={`${planHeading} pt-3`} data-aos="fade-up">
        {title}
      </div>
      <div
        className={`${subText2} py-0 py-sm-0 py-md-0 py-lg-0 py-xl-2`}
        data-aos="fade-up"
      >
        {description}
      </div>
      <div className="position-absolute bottom-0 pb-3" data-aos="fade-up">
        {!isCTAExternal ? (
          <Link to={ctaLink}>
            <label className={`${cta1} pe-auto cursorPointer`} role="link">
              {ctaLabel}
              <span className={`${ctaIconPos}`}>
                <img
                  className={`${ctaIcon} ps-2`}
                  src={forwardArrow}
                  alt={ctaLabel}
                />
              </span>
            </label>
          </Link>
        ) : (
          <a href={ctaLink} rel="external" target="_blank">
            <label className={`${cta1} pe-auto cursorPointer`} role="link">
              {ctaLabel}
              <span className={`${ctaIconPos}`}>
                <img
                  className={`${ctaIcon} ps-2`}
                  src={forwardArrow}
                  alt={ctaLabel}
                />
              </span>
            </label>
          </a>
        )}
      </div>
    </div>
  );
};
