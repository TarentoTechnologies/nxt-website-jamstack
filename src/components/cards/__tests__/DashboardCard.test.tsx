import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import React from "react";

import { DashboardCard } from "../DashboardCard";

const imgData = {
  localFile: {
    childImageSharp: {
      gatsbyImageData: {
        layout: "constrained",
        placeholder: {
          fallback:
            "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAAUCAYAAACNiR0NAAAACXBIWXMAAAsTAAALEwEAmpwYAAADVElEQVR42o1V229MQRw+5akRBOkrCXFpUZRSlXiQtAlpCJE0jVulIfEHEJ5F3OKBukXE7QGNjQjxzkuDaLtF6xLt2XP20na7drslQXfmfH7fnHNY1GWS2Z2dne833/x+33xjWUEDUCJ9Isd9fZgai6ltrqMirqNd+VbSPddVtvyOxB3sdF2UBriJxFq/tnDScdSuuKsHR/NAfgTe4AAQdz3pGhxz7mOec6o/4ajGYmzYSgJ2JXZ/oVUASA/BE0Y6ZMZg7IalP6eH0/ByWcC29RlY+B6D0Sf4zPS5L5/NzmNOTKkwyJ+6EysoriWG2ICpiWW5tmribrJASTBPdkcuB3zIANkP/nfxmGu5hmsTcV0YyTGoajbBolFMGkjpd3JMyJE1Aedah7BkURdqV71A9bJurFrRjZrqbqys6jZzi8u7cPb0EDLDQMwuaOZWYiR6ezHDSiSwm3njMWU3c5yK+R04uN9B5cIuNO94j00Nr7G+vhd7WvrMRgcPOFgwrwNxR4MYYskylcJ2K5nQEf6QnRSPdO/uCMrndYKVJJvHj0Zx5HAShyTIs6efsLb2JaiAReWdaLuVNWkgljGSSX3HkorFSFmO65HppQtpLBUWDx/kMWVyO65cHkbDhl7U1/XgxrUMJpW24/69PFYsjZpjE0MsY0isuBXKwg3o27bCwvmdKJv+FFs3v8WcWc+xpuYl6tb1YPbM59iy8Q3Kpj1B+dwOCaQMhli/+kpZ/PghBWUqWL0silMnBzD2Feh59ZULkUpqGY+BMmHRVkuReFxiinVKhvGArsedhqVyVZVRXDyfDgA+AwI5zogKrl/NoLoqaqTD/4gNYsQsJpIJ9W+ANlJYviRq8sWWy3qmCOwcs0XasqgRKf0IGBRFCmyx1GFAMqAOKY2W3e9NFS9fSpvCsHN8+2YW+/aKfERS3DxgqFgcStCiGClKUo7Zckdl0YljKQOgkCns4s65yoouHD+aMumhsHkpeDl4SazAYZrJUhJbYD64M/P3t86T+ApRyjcJ1WQVX+ifzaHwH+agxjeH0HZoQbSi4OL/075ocYGoW79b13gGm3DQKOD+jz8ZrDYmGxrsqDFYPUgzHs9gf3sCaO/G5sXufdsPmZnnIMLngc/EeE/AN37w0Ej7/leXAAAAAElFTkSuQmCC",
        },
        images: {
          fallback: {
            src: "/static/f9e009176138934da4f317f3db96c6ee/e9fba/Group_11_0e80584bfd.png",
            srcSet:
              "/static/f9e009176138934da4f317f3db96c6ee/86ee2/Group_11_0e80584bfd.png 13w,\n/static/f9e009176138934da4f317f3db96c6ee/88208/Group_11_0e80584bfd.png 25w,\n/static/f9e009176138934da4f317f3db96c6ee/e9fba/Group_11_0e80584bfd.png 50w",
            sizes: "(min-width: 50px) 50px, 100vw",
          },
          sources: [],
        },
        width: 50,
        height: 50,
      },
    },
  },
  alternativeText: "Retail",
};

describe("Test if the DashboardCard component is working as expected", () => {
  it("<DashboardCard /> matches snapshot", () => {
    const component = render(
      <DashboardCard
        title={"Retail"}
        description={
          "Effortlessly monitor retail performance with our interactive dashboard cards on your website."
        }
        imgSrc={imgData}
        ctaLabel={"Explore"}
        ctaLink={"/rain/demo"}
        isCTAExternal={true}
      />
    );
    expect(component.container).toMatchSnapshot();
  });

  it("<DashboardCard /> renders without exploding", () => {
    const component = render(
      <DashboardCard
        title={"Retail"}
        description={
          "Effortlessly monitor retail performance with our interactive dashboard cards on your website."
        }
        imgSrc={imgData}
        ctaLabel={"Explore"}
        ctaLink={"/rain/demo"}
        isCTAExternal={true}
      />
    );
    expect(component.getByTestId("dashboard-card")).toBeInTheDocument();
  });
});
