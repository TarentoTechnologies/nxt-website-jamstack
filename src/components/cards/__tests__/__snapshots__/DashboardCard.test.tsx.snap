// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if the DashboardCard component is working as expected <DashboardCard /> matches snapshot 1`] = `
<div>
  <div
    class="dashboardCard whiteBg position-relative"
    data-testid="dashboard-card"
  >
    <div
      data-aos="fade-up"
    >
      <div
        class="gatsby-image-wrapper gatsby-image-wrapper-constrained"
        data-gatsby-image-wrapper=""
        style="position: relative; overflow: hidden; display: inline-block; vertical-align: top;"
      >
        <div
          style="max-width: 50px; display: block;"
        >
          <img
            alt=""
            aria-hidden="true"
            role="presentation"
            src="data:image/svg+xml;charset=utf-8,%3Csvg%20height='50'%20width='50'%20xmlns='http://www.w3.org/2000/svg'%20version='1.1'%3E%3C/svg%3E"
            style="max-width: 100%; display: block; position: static;"
          />
        </div>
        <img
          alt=""
          aria-hidden="true"
          data-placeholder-image=""
          decoding="async"
          src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAAUCAYAAACNiR0NAAAACXBIWXMAAAsTAAALEwEAmpwYAAADVElEQVR42o1V229MQRw+5akRBOkrCXFpUZRSlXiQtAlpCJE0jVulIfEHEJ5F3OKBukXE7QGNjQjxzkuDaLtF6xLt2XP20na7drslQXfmfH7fnHNY1GWS2Z2dne833/x+33xjWUEDUCJ9Isd9fZgai6ltrqMirqNd+VbSPddVtvyOxB3sdF2UBriJxFq/tnDScdSuuKsHR/NAfgTe4AAQdz3pGhxz7mOec6o/4ajGYmzYSgJ2JXZ/oVUASA/BE0Y6ZMZg7IalP6eH0/ByWcC29RlY+B6D0Sf4zPS5L5/NzmNOTKkwyJ+6EysoriWG2ICpiWW5tmribrJASTBPdkcuB3zIANkP/nfxmGu5hmsTcV0YyTGoajbBolFMGkjpd3JMyJE1Aedah7BkURdqV71A9bJurFrRjZrqbqys6jZzi8u7cPb0EDLDQMwuaOZWYiR6ezHDSiSwm3njMWU3c5yK+R04uN9B5cIuNO94j00Nr7G+vhd7WvrMRgcPOFgwrwNxR4MYYskylcJ2K5nQEf6QnRSPdO/uCMrndYKVJJvHj0Zx5HAShyTIs6efsLb2JaiAReWdaLuVNWkgljGSSX3HkorFSFmO65HppQtpLBUWDx/kMWVyO65cHkbDhl7U1/XgxrUMJpW24/69PFYsjZpjE0MsY0isuBXKwg3o27bCwvmdKJv+FFs3v8WcWc+xpuYl6tb1YPbM59iy8Q3Kpj1B+dwOCaQMhli/+kpZ/PghBWUqWL0silMnBzD2Feh59ZULkUpqGY+BMmHRVkuReFxiinVKhvGArsedhqVyVZVRXDyfDgA+AwI5zogKrl/NoLoqaqTD/4gNYsQsJpIJ9W+ANlJYviRq8sWWy3qmCOwcs0XasqgRKf0IGBRFCmyx1GFAMqAOKY2W3e9NFS9fSpvCsHN8+2YW+/aKfERS3DxgqFgcStCiGClKUo7Zckdl0YljKQOgkCns4s65yoouHD+aMumhsHkpeDl4SazAYZrJUhJbYD64M/P3t86T+ApRyjcJ1WQVX+ifzaHwH+agxjeH0HZoQbSi4OL/075ocYGoW79b13gGm3DQKOD+jz8ZrDYmGxrsqDFYPUgzHs9gf3sCaO/G5sXufdsPmZnnIMLngc/EeE/AN37w0Ej7/leXAAAAAElFTkSuQmCC"
          style="height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
        />
        <img
          alt=""
          data-gatsby-image-ssr=""
          data-main-image=""
          data-src="/static/f9e009176138934da4f317f3db96c6ee/e9fba/Group_11_0e80584bfd.png"
          data-srcset="/static/f9e009176138934da4f317f3db96c6ee/86ee2/Group_11_0e80584bfd.png 13w,/static/f9e009176138934da4f317f3db96c6ee/88208/Group_11_0e80584bfd.png 25w,/static/f9e009176138934da4f317f3db96c6ee/e9fba/Group_11_0e80584bfd.png 50w"
          decoding="async"
          loading="lazy"
          sizes="(min-width: 50px) 50px, 100vw"
          style="height: 100%; left: 0px; position: absolute; top: 0px; transform: translateZ(0); transition: opacity 250ms linear; width: 100%; will-change: opacity; opacity: 0;"
        />
        <noscript />
        <script
          type="module"
        >
          const t="undefined"!=typeof HTMLImageElement&&"loading"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll("img[data-main-image]");for(let e of t){e.dataset.src&&(e.setAttribute("src",e.dataset.src),e.removeAttribute("data-src")),e.dataset.srcset&&(e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset"));const t=e.parentNode.querySelectorAll("source[data-srcset]");for(let e of t)e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset");e.complete&&(e.style.opacity=1,e.parentNode.parentNode.querySelector("[data-placeholder-image]").style.opacity=0)}}
        </script>
      </div>
    </div>
    <div
      class="planHeading pt-3"
      data-aos="fade-up"
    >
      Retail
    </div>
    <div
      class="subText2 py-0 py-sm-0 py-md-0 py-lg-0 py-xl-2"
      data-aos="fade-up"
    >
      Effortlessly monitor retail performance with our interactive dashboard cards on your website.
    </div>
    <div
      class="position-absolute bottom-0 pb-3"
      data-aos="fade-up"
    >
      <a
        href="/rain/demo"
        rel="external"
        target="_blank"
      >
        <label
          class="cta1 pe-auto cursorPointer"
          role="link"
        >
          Explore
          <span
            class="ctaIconPos"
          >
            <img
              alt="Explore"
              class="ctaIcon ps-2"
              src="test-file-stub"
            />
          </span>
        </label>
      </a>
    </div>
  </div>
</div>
`;
